File: water-backend/app/Models/Vendor.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Vendor extends Model
{
    use HasFactory;

    protected $fillable = ['user_id', 'business_name', 'address', 'lat', 'lng', 'service_radius_km', 'is_open'];
    protected $casts    = ['is_open' => 'boolean', 'lat' => 'float', 'lng' => 'float', 'service_radius_km' => 'float'];

    public function user(): \Illuminate\Database\Eloquent\Relations\BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function products(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(Product::class);
    }

    public function orders(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(Order::class);
    }

    public function deliveryBoys(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(VendorDeliveryBoy::class);
    }

    public function reviews(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(Review::class);
    }
}

File: water-app/lib/shared/models/vendor.dart
import 'package:equatable/equatable.dart';

class Vendor extends Equatable {
  final int id;
  final int userId;
  final String businessName;
  final String address;
  final double lat;
  final double lng;
  final double serviceRadiusKm;
  final bool isOpen;
  final double? distance;

  const Vendor({
    required this.id,
    required this.userId,
    required this.businessName,
    required this.address,
    required this.lat,
    required this.lng,
    required this.serviceRadiusKm,
    required this.isOpen,
    this.distance,
  });

  factory Vendor.fromJson(Map<String, dynamic> json) => Vendor(
    id:              json['id'] as int,
    userId:          json['user_id'] as int,
    businessName:    json['business_name'] as String,
    address:         json['address'] as String,
    lat:             (json['lat'] as num).toDouble(),
    lng:             (json['lng'] as num).toDouble(),
    serviceRadiusKm: (json['service_radius_km'] as num).toDouble(),
    isOpen:          json['is_open'] as bool,
    distance:        json['distance'] != null ? (json['distance'] as num).toDouble() : null,
  );

  @override
  List<Object?> get props => [id, businessName, isOpen];
}
